Data/Primitive/PrimArray.hs

```diff
@@ -467,6 +469,16 @@ sizeofPrimArray :: forall a. Prim a => PrimArray a -> Int
 {-# INLINE sizeofPrimArray #-}
 sizeofPrimArray (PrimArray arr#) = I# (quotInt# (sizeofByteArray# arr#) (sizeOf# (undefined :: a)))
 
+-- | Lazily map each element of the primitive array to a monoid, and combine the results.
+foldMapPrimArray :: forall a m. (Prim a, Monoid m) => (a -> m) -> PrimArray a -> m
+{-# INLINE foldMapPrimArray #-}
+foldMapPrimArray f = foldrPrimArray (\a acc -> acc `mappend` f a) mempty
+
+-- | Strictly map each element of the primitive array to a monoid, and combine the results.
+foldMapPrimArray' :: forall a m. (Prim a, Monoid m) => (a -> m) -> PrimArray a -> m
```

Comment:
The documentation is insufficient. It should explain which way the combination associates. Also, it's notably not (necessarily) strict in the function results. The documentation should reflect this and there should be a comment explaining that decision.